lib/syskit/gui/logging_configuration.rb

```diff
@@ -0,0 +1,92 @@
+require 'vizkit'
+require 'vizkit/vizkit_items'
+require 'vizkit/tree_view'
+require 'Qt4'
+require 'syskit/shell_interface'
+require 'syskit/gui/logging_configuration_item'
+require 'roby/interface/exceptions'
+
+module Syskit
+    module GUI
+        # A widget containing an editable TreeView to allow the user to
+        # manage basic Syskit's logging configuration
+        class LoggingConfiguration < Qt::Widget
+            attr_reader :model, :treeView, :syskit, :item_name, :item_value
+            def initialize(parent = nil, syskit)
+                super(parent)
+                main_layout = Qt::VBoxLayout.new(self)
+                @treeView = Qt::TreeView.new
+
+                Vizkit.setup_tree_view treeView
+                @model = Vizkit::VizkitItemModel.new
+                treeView.setModel @model
+                main_layout.add_widget(treeView)
+                treeView.setColumnWidth(0, 200)
+                treeView.style_sheet = "QTreeView { background-color: rgb(255, 255, 219);
+                                                    alternate-background-color: rgb(255, 255, 174);
+                                                    color: rgb(0, 0, 0); }
+                                        QTreeView:disabled { color: rgb(159, 158, 158); }"
+
+                @syskit = syskit
+                @timer = Qt::Timer.new
+                @timer.connect(SIGNAL('timeout()')) { refresh }
+                @timer.start 10000
+
+                refresh
+            end
+
+            # Fetches the current logging configuration from syskit's
+            # sync interface
+            def refresh
+                if !syskit.client.nil?
```

Comment:
This is `if syskit.reachable?`. We unfortunately still have to use `syskit.client` directly in some cases, but I'm trying to remove that need as much as possible.

Reply:
👍  (I have squashed the patch for this one)